### PR TITLE
Adding guidance for referencing man pages

### PR DESCRIPTION
**Pull request by bergerhoffer**
Draft for addressing #224.

Preview: https://file.rdu.redhat.com/~ahoffer/2024/main-man-pages.html#man-pages

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@redhat-documentation/ccs-style-council Draft is ready for adding guidance for how to reference man pages. 

My only outstanding questions are:

1. Is it okay to add something to the additional resources sections that is not a link? Or is a link a true requirement? @emmurphy1 is this something you can weigh in on?
2. Thoughts on if "Formatting" is the right section that this should go under?

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (mportman12) [inline supplementary_style_guide/style_guidelines/formatting.adoc:14]:
```suggestion
When referencing a man page, use the following format: _`<name_of_utility>(<section_number>)` man page_.
```
Italicize the template text to make it clear what is part of the template and what isn't. This formatting is used for date formats.

**Review by mportman12 — CHANGES_REQUESTED** [review]
Formatting section is a good location for this. Left a formatting recommendation, otherwise looks good.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/formatting.adoc]: Thanks, updated!

Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/formatting.adoc]: Actually one sec, that messed up the backtick formatting.

Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/formatting.adoc]: Okay, updated

**Review by bburt-rh — DISMISSED** [review]

**Review by mportman12 — DISMISSED** [review]
LGTM

**Review by ncbaratta — APPROVED** [review]
lgtm

**Comment by emmurphy1** [comment]
@bergerhoffer   I checked the templates and ref guide and we don't say that you can reference man pages in additional resources, but I think they should be allowed. Let me get a discussion going on the the mod docs channel and I'll update here.

**Comment by bergerhoffer** [comment]
Putting this PR on hold until we hear the result from the mod docs team!

**Comment by soulcharmer** [comment]
I have one more issue to discuss.

When referencing a man page in the docs on the CP, the page is not readily available to the user. It appears that often, the user concludes that they need to google the man page, especially if they just browse the docs and do not have a subscription yet. We even got several bug reports to that effect, most recently this one: https://bugzilla.redhat.com/show_bug.cgi?id=2158027 This is a gap and the UX is also not that great because it is reasonable to expect to have the man page readily available on the CP.

However, in some products, there are big technical and logistical challenges to generate a stand-alone man page portal that would close this gap.

At the same time, the man page references are actually meant to point to the ones the user has installed on their system. If this detail is not explicit in the docs though, it can lead to bug reports or affect the UX...

A fairly simple workaround could be to set up a guideline to refer to man pages in the docs in a certain way and with the above in mind.

Thus far, we brainstormed two possible solutions in our team:

1) mention the installed system explicitly in the reference snippet, for example "For details, see the <xyz> man page on your installed system.

2) do not reference the man page alone but a command that will display it, implying that accessing a running system should be the next step, for example "For details, run the 'man <xyz>' (on your system)"


Perhaps there are even more elegant or user-friendly alternatives to achieve the same effect.

Please review this proposal and let me know if it viable for the SSG.

Thank you

**Comment by mmuehlfeldRH** [comment]
>1. mention the installed system explicitly in the reference snippet, for example "For details, see the [xyz] man page on your installed system.
>
>2. do not reference the man page alone but a command that will display it, implying that accessing a running system should be the next step, for example "For details, run the 'man [xyz]' command (on your system)"

In my opinion, both options are not optimal, because we should keep additional resources entries as short as possible. I was told to remove "For details,..." and similar from all my additional resources entries. (I forgot if this was mostly because of the table format of additional resources in Pv2).

I would vote for a variant of option 1:

* `sudoers(5)` man page on your installed system

or even shorter:

* `sudoers(5)` man page on the system

For me it would be clear that I don't run the procedure on one host and then try to open the referenced man page on a different host. WDYT?

My argument against the version with the command is that, in the rest of our modules, commands are (in most cases) shown in text boxes and not inline.

**Comment by dfitzmau** [comment]
Note: Hold off merging this PR until the `On hold` label has been removed. The label in this case flags that a discussion is still in progress.

**Comment by soulcharmer** [comment]
Right, I do not have a strong preference on how exactly the reference snippet should look like, as long as it acknowledges the ambiguity of plainly naming the man page in the docs while not making it readily available on the same platform.

I'd agree that using the "man [xyz]" commands is a bit of a roundabout solution and I'm leaning in the direction of using a simple suffix to indicate the intended location of the man page.

Shall we collect more opinions and/or approvals on this guidance?

**Comment by bredamc** [comment]
Updated the original [issue](https://github.com/redhat-documentation/supplementary-style-guide/issues/224) to suggest adding a general module to explain common info about man pages (details in the issue). And then we could link to that explanatory module whenever we refer to a man page?

**Comment by Srivaralakshmi** [comment]
@bergerhoffer and @bredamc Any progress on this PR?

**Comment by bergerhoffer** [comment]
I've updated the PR to add "on your system" to address the latest comments here.

I also added a sentence mentioning that users run the "man" command to view the information.

The only last thing is just making sure that it is acceptable to have something that is not a link in an additional resources section. @emmurphy1 any updates on this from the mod docs side?

**Comment by bredamc** [comment]
@emmurphy1 @bergerhoffer Ping :) 

@emmurphy1 Can you please confirm whether the mod doc group agreed that it is acceptable to include non-linked text in the Additional resources section?

@bergerhoffer Should we include an example that lists multiple man pages, to show whether to include them on the same line or multiple lines?

**Comment by mmuehlfeldRH** [comment]
> Should we include an example that lists multiple man pages, to show whether to include them on the same line or multiple lines?

+1 for either adding an example or spelling out to list them on the same/multiple lines.


Some additional thoughts from the RHEL admin perspectve:

TL;DR: Man pages and local docs are typically mentioned in our docs for reference purposes (not needed to understand the procedure but they provide additional information, etc.). For this reason, they fit better in additional resources than mentinging x man pages in x different places in the procedure.

Most things on RHEL servers happen on the command line. Tools, config files of services, etc. - they all ship at least one man page (Systemd ships meanwhile 227 and they are great resources!). They all provide a big benefit for admins and they are a great addition to our docs.

In my docs, I explain config options in a way that it's clear what they do, but I often link the corresponding man page in additional resources, because it's a great reference for further details about the options or to figure out other config options that might be helpful for the admin in their own environment. ([Example](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/configuring_and_managing_networking/assembly_setting-up-a-wireguard-vpn_configuring-and-managing-networking#proc_configuring-a-wireguard-by-using-nmcli_assembly_setting-up-a-wireguard-vpn))

Many packages also install readme files, example configs, etc. in /usr/share/docs/ ([Example](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/deploying_mail_servers/configuring-and-maintaining-a-dovecot-imap-and-pop3-server_deploying-mail-servers#preparing-dovecot-to-use-virtual-users_setting-up-a-dovecot-server-with-pam-authentication)). They are also a great help in many cases and worth referencing them in additional resources - even if they also cannot be linked.

Man pages and local docs are a great help and reference, and they save us from duplicating a lot of information in our guides - at least in RHEL where admins typically work on the command line.

**Comment by bredamc** [comment]
Thanks, Marc! As mentioned in the latest comment in the related Issue 224:
"Discussed at April SSG meeting. Group agreed that we should recommend but not require that man pages be listed on one bullet in a comma-delimited list. @bergerhoffer can you update the PR for this decision?"

**Comment by bergerhoffer** [comment]
Apologies for the delay on updating this. 

Updated to add an example for multiple man pages. I also updated "It is not necessary to link to a website..." to "Do not link to a website...".

Please take another look when you get a chance.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [inline supplementary_style_guide/style_guidelines/formatting.adoc:14]: 1. Should we highlight "Additional resources" in some way?
2. Would the text be easier to read if the format was on a new line?

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [inline supplementary_style_guide/style_guidelines/formatting.adoc:16]: Should we explain that the man-page content can vary from one system to another, and the user must run the `man` command on the system where the named command will be run, to ensure that they see the relevant system-specific information?

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (mmuehlfeldRH) [reply supplementary_style_guide/style_guidelines/formatting.adoc:14]: 1. Technical nitpick: `<name_of_utility>` is only correct for section 8 and 1. There are also man pages for config files `NetworkManager.conf(5)`, system calls `exit(2)`, libraries `exit(3)`, and others . I would recommend `<man_page_name>(<section_number>)`
2. Is there a special reason for italic on "man page on your system"? The [ISG](https://www.ibm.com/docs/en/ibm-style?topic=references-citations#general-guidelines) does not use italic for qualifiers (`For more information, see the _DQTI_ book.`)

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [reply supplementary_style_guide/style_guidelines/formatting.adoc:16]: I meant that I think it's useful to explain _why_ we're including the text "on your system" in the examples.

Comment (mmuehlfeldRH) [reply supplementary_style_guide/style_guidelines/formatting.adoc:16]: If it's about adding the why to SSG (not to the actual entries), I agree that it would be helpful.

**Comment by mportman12** [comment]
@bergerhoffer PTAL at outstanding review comments.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (libander) [reply supplementary_style_guide/style_guidelines/formatting.adoc:16]: If we're adding the why to the SSG, it is important to note that the content of man files is determined not by OS version, but by kernel version. You can have multiple kernel versions across the life of an OS Version.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (mmuehlfeldRH) [reply supplementary_style_guide/style_guidelines/formatting.adoc:14]: @libander I don't understand your comment. How is the `man` man page content you posted relevant for the proposed change in the SSG?

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (mmuehlfeldRH) [reply supplementary_style_guide/style_guidelines/formatting.adoc:16]: @libander what do you mean with "the content of man files is determined not by OS version, but by kernel version"?

Man pages are shipped with the RPMs of tools, services, etc. they belong to. For example, `find(1)` is provided by the `findutils` RPM, and `rsync(1)` by the `rsync` package. If you update one of these packages, then the man pages are also updated (but the man page content only changes if the new RPM also contains a different man page). Anyway, man pages don't change if you boot a different kernel.

None of the kernel packages (`kernel`, `kernel-core`, `kernel-modules`, `kernel-modules-extra`, `kernel-modules-core`) contains a man page. I don't know if there are other packages than `kernel*` that can be installed at the same time in different versions, but if such packages contain man pages, then the old man pages are replaced by the new ones because the files use the same paths to the man page files. If you check, for example, `/usr/share/man/man1/` you see that there's only one man page for each tool.

In case that you meant something else, can you please explain it further?

Comment (libander) [reply supplementary_style_guide/style_guidelines/formatting.adoc:16]: > In case that you meant something else, can you please explain it further?

You're right, many packages come with their own man-page included. The [`man-pages.rpm`](https://access.redhat.com/downloads/content/rhel---9/x86_64/11051/man-pages/6.04-1.el9/src/fd431d51/package) provides man-pages for [GNU/Linux](https://git.kernel.org/pub/scm/docs/man-pages/man-pages.git/about/) system and library calls, etc. 

> adding the why to SSG (not to the actual entries)

It was with this context in mind I was suggesting that we be slightly more specific than 'vary by system'  -- 'vary by package or kernel version' would be less mysterious. :)

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (libander) [reply supplementary_style_guide/style_guidelines/formatting.adoc:14]: > 1. Technical nitpick: `<name_of_utility>` is only correct for section 8 and 1. There are also man pages for config files `NetworkManager.conf(5)`, system calls `exit(2)`, libraries `exit(3)`, and others . I would recommend `<man_page_name>(<section_number>)`

Likewise, given that the man pages themselves classify the pages: 
> Each page argument given to man is normally the name of a program, utility or function.

I'd consider changing line 16's: 
`to view the information for the named command or utility.` 
to: 
`to view the information for the program, utility or function.`

**Comment by mportman12** [comment]
@bergerhoffer Please take a look at outstanding review comments.

**Comment by bergerhoffer** [comment]
Updated per the latest feedback. PTAL: https://file.rdu.redhat.com/~ahoffer/2024/main-man-pages.html#man-pages

**Comment by bredamc** [comment]
Many thanks for the preview link, @bergerhoffer! Very helpful :)

Not sure if you didn't see my comments of 27 Sept or are choosing to not implement them, so I'll reiterate them here, just in case :)

1.     Should we highlight "Additional resources" in some way?
2.     Would the text be easier to read if the format was on a new line?

I'll also add that I think we should not use italics for the text "man page on your system", because that text is to be reproduced verbatim.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [inline supplementary_style_guide/style_guidelines/formatting.adoc:14]:
```suggestion
When referencing a man page in an "Additional resources" section, use the following format: 
* `_<man_page_name>(<section_number>)_` man page on your system
```

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [inline supplementary_style_guide/style_guidelines/formatting.adoc:16]:
```suggestion
Do not link to a website that contains the man page information. 
The contents of a man page might vary between systems or package versions, so users must run the `man` command on the target system to view the system-specific information for the named command or utility.
```

**Review by bredamc — COMMENTED** [review]
A few minor comments.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/formatting.adoc:14]: No sorry, I missed this suggestion! I agree that this looks better, will update.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/formatting.adoc:16]: Thanks! I used your wording suggestions, but kept them on the same line (because the info about users having to run the `man` command on their system is a reason for why we say not to link to man page websites)

**Comment by bergerhoffer** [comment]
@bredamc sorry for missing your suggestions earlier! Updated if you can please take another look

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [reply supplementary_style_guide/style_guidelines/formatting.adoc]: Putting them on separate lines was just to make it easier to review (each sentence on a separate line) -- it didn't affect the output, the two sentences were still in the same paragraph. But it was a tiny suggestion, no problem at all for not implementing it :)

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [inline supplementary_style_guide/style_guidelines/formatting.adoc:16]:
```suggestion
* `_<man_page_name>_(_<section_number>_)` man page on your system
```
The text "man page on your system" should not be in italics.  It's confusing when it's in italics, because it seems as if it should be replaced by something else, when in fact it should be reproduced verbatim.

The parentheses shouldn't be in italics either, but that's not as bothersome :)

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/formatting.adoc]: Gotcha - yeah I normally don't write my sentences that way like I know some people do :)

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/formatting.adoc]: Thanks, updated!

**Comment by bergerhoffer** [comment]
@bredamc Updated, ready for another look! Didn't update the preview yet though because something is erroring out when I try the transfer

**Comment by Srivaralakshmi** [comment]
@bredamc Can you please take another look at this revised PR? TY!

**Review by bredamc — APPROVED** [review]
LGTM -- thanks, @bergerhoffer!
(And thanks for the reminder to approve, @Srivaralakshmi!)

**Comment by Srivaralakshmi** [comment]
@bburt-rh , @mportman12 , @ncbaratta Need two more approvals. Please help. TY!

**Review by mportman12 — APPROVED** [review]
LGTM

**Comment by Srivaralakshmi** [comment]
Thank you all!

**Pull request merged**
